compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -3117,9 +3117,10 @@ uint8_t OMR::X86::CodeGenerator::nodeResultGPRCount(TR::Node *node, TR_RegisterP
    TR_ASSERT(!self()->comp()->getOption(TR_DisableRegisterPressureSimulation), "assertion failure");
 
    // 32-bit integer constants practically never need a register on x86
+   //  (includes 64-bit integer constants with high word zero needed to maintain IL correctness)
    //
    if (  node->getOpCode().isLoadConst()
-      && node->getSize() <= 4
+      && (node->getSize() <= 4 || node->getType().isInt64() && node->isHighWordZero())
```

Comment:
May I suggest parenthesis around the `node->getType().isInt64() && node->isHighWordZero()`? A keen reader will correctly parse this however mere mortals may not remember that `&&` has higher precedence than `||`  in C++ and may interpret this line incorrectly.